js/src/css/beautifier.js

```diff
@@ -92,11 +92,15 @@ Beautifier.prototype.eatString = function(endChars) {
 Beautifier.prototype.eatWhitespace = function(allowAtLeastOneNewLine) {
   var result = whitespaceChar.test(this._input.peek());
   var isFirstNewLine = true;
-
+  var newlines = this._options.max_preserve_newlines;
   while (whitespaceChar.test(this._input.peek())) {
     this._ch = this._input.next();
     if (allowAtLeastOneNewLine && this._ch === '\n') {
-      if (this._options.preserve_newlines || isFirstNewLine) {
+      if ((this._options.preserve_newlines || isFirstNewLine) && (this._options.max_preserve_newlines === 32786 || this._options.max_preserve_newlines === 0)) {
+        isFirstNewLine = false;
+        this._output.add_new_line(true);
+      } else if ((this._options.preserve_newlines || isFirstNewLine) && newlines > 0) {
+        newlines--;
```

Comment:
max_preserve_newlines is already sanitized by the options object:

https://github.com/beautify-web/js-beautify/blob/aa9e365aed991080193c2bced411f132990c9d82/js/src/core/options.js#L44-47

```suggestion
      if (newline_count === 0 || newline_count < this._options.max_preserve_newlines) {
        newline_count++;
```

Also delete `isFirstNewLine = false;` below.

And implement same in python, of course.

Reply:
Yes I liked this way better, is way cleaner, thank you!